legend-depot-store-mongo/src/main/java/org/finos/legend/depot/store/mongo/admin/metrics/QueryMetricsMongo.java

```diff
@@ -66,7 +66,7 @@ public List<VersionQueryCounter> get(String groupId, String artifactId, String v
     }
 
     @Override
-    public void persistMetrics(List<VersionQueryCounter> metrics)
+    public synchronized void persistMetrics(List<VersionQueryCounter> metrics)
```

Comment:
dont think this will help/work
what u need is to make sure that when the persists metrics starts:
- get hold of the existing metrics collection in a variable 
- flush : create a new empty stats collection and set metrics .variable
- then  persist the old collection 

we need tests for this of course

Reply:
since this metrics is a synchronized list, which means for each write there would be a lock and for iteration in reading we need to make it synchronized as I looked into the documentation